### PR TITLE
Mongooseimctl http upload

### PR DESCRIPTION
**Pull request by DenysGonchar**
1. fixed x-amz-acl parameter passsing. It must be added as signed header, for more details see [this minio issue](https://github.com/minio/minio/issues/5083) and [spec](https://docs.aws.amazon.com/AmazonS3/latest/API/sigv4-query-string-auth.html):

   - Canonical Headers must include the HTTP host header. If you plan to include any of the x-amz-* headers, these headers must also be added for signature calculation. You can optionally add all other headers that you plan to include in your request. For added security, you should sign as many headers as possible. 

2. changed `add_acl` default value to `false`, as it's not supported by minio at all.

3. added `mongooseimctl http_upload` command to generate upload/download links (`mod_http_upload` must be configured and `upload` must be added to the `service_admin_extra` list in `mongooseim.cfg`). generated links can be tested like that:

```bash
# create some text file
echo qwerty > tmp.txt
# get file size
filesize="$(wc -c tmp.txt | awk '{print $1}')"
# generate upload/download urls
urls="$(./mongooseimctl http_upload localhost tmp.txt "$filesize" "text/plain" 600)"
put_url="$(echo "$urls" | awk '/PutURL:/ {print $2}')"
get_url="$(echo "$urls" | awk '/GetURL:/ {print $2}')"
# upload a file, note the if add_acl is enabled then
# you must also add the next curl option:
#    -H "x-amz-acl: public-read"
curl -v -T "./tmp.txt" -H "Content-Type: text/plain" "$put_url"
# try to download a file
curl -i "$get_url"
```

### CODE REVIEW

**Comment by mongoose-im** [comment]
[8123.1](https://travis-ci.org/esl/MongooseIM/jobs/672356805) / Erlang 22.0 / small_tests / 42b6ee261cf7000bf603d71cbd801f6fb8fccbf0
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2708/8123/small_tests.22.0) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2708/8123/small_tests.22.0/small/ct_run.mongooseim@localhost.2020-04-08_03.57.21/lib.mongooseim.logs/run.2020-04-08_03.57.22/suite.log.html)

File: src/admin_extra/service_admin_extra_upload.erl [hunk 1]
```diff
@@ -0,0 +1,54 @@
+-module(service_admin_extra_upload).
```
Comment (michalwski) [inline src/admin_extra/service_admin_extra_upload.erl:1]: As far as I can tell there are no tests for these new ctl commands. It'd be really good to have them added. There suite `big_tests/tests/ejabberdctl_SUITE.erl` where such tests could be added. They can also be added to the  `big_tests/tests/mod_http_upload_SUITE.erl`

**Comment by mongoose-im** [comment]
[8125.1](https://travis-ci.org/esl/MongooseIM/jobs/672428326) / Erlang 22.0 / small_tests / 2b155557161f32f4db2b1ea5c4fbf348cc7b4c26
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2708/8125/small_tests.22.0) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2708/8125/small_tests.22.0/small/ct_run.mongooseim@localhost.2020-04-08_09.26.20/lib.mongooseim.logs/run.2020-04-08_09.26.20/suite.log.html)

**Comment by mongoose-im** [comment]
[8129.1](https://travis-ci.org/esl/MongooseIM/jobs/672953633) / Erlang 22.0 / small_tests / c62647b82c89a6ad22fb83943b095bec379f5645
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2708/8129/small_tests.22.0) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2708/8129/small_tests.22.0/small/ct_run.mongooseim@localhost.2020-04-09_12.43.45/lib.mongooseim.logs/run.2020-04-09_12.43.46/suite.log.html)

**Comment by mongoose-im** [comment]
[8131.1](https://travis-ci.org/esl/MongooseIM/jobs/673190809) / Erlang 22.0 / small_tests / 2c960169e27f7ebe511a7c94a223868f013eb377
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2708/8131/small_tests.22.0) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2708/8131/small_tests.22.0/small/ct_run.mongooseim@localhost.2020-04-09_22.05.18/lib.mongooseim.logs/run.2020-04-09_22.05.19/suite.log.html)

**Comment by mongoose-im** [comment]
[8132.1](https://travis-ci.org/esl/MongooseIM/jobs/673206137) / Erlang 22.0 / small_tests / 26bd0e890ca0b88dbeecd6e76ad917ca54fd5655
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2708/8132/small_tests.22.0) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2708/8132/small_tests.22.0/small/ct_run.mongooseim@localhost.2020-04-09_22.59.22/lib.mongooseim.logs/run.2020-04-09_22.59.22/suite.log.html)

**Comment by mongoose-im** [comment]
[8133.1](https://travis-ci.org/esl/MongooseIM/jobs/673210117) / Erlang 22.0 / small_tests / c07c4384bb08d450636a9d664e30834a271ebb23
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2708/8133/small_tests.22.0) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2708/8133/small_tests.22.0/small/ct_run.mongooseim@localhost.2020-04-09_23.28.53/lib.mongooseim.logs/run.2020-04-09_23.28.54/suite.log.html)

**Comment by mongoose-im** [comment]
[8134.1](https://travis-ci.org/esl/MongooseIM/jobs/673251519) / Erlang 22.0 / small_tests / bd3e1fa2d831889d1e7f6ca95b9a5238c9497352
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2708/8134/small_tests.22.0) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2708/8134/small_tests.22.0/small/ct_run.mongooseim@localhost.2020-04-10_01.51.52/lib.mongooseim.logs/run.2020-04-10_01.51.52/suite.log.html)

**Comment by mongoose-im** [comment]
[8135.1](https://travis-ci.org/esl/MongooseIM/jobs/673252428) / Erlang 22.0 / small_tests / 13b44c4c7f3b47f40288c86bc26245673718855b
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2708/8135/small_tests.22.0) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2708/8135/small_tests.22.0/small/ct_run.mongooseim@localhost.2020-04-10_02.19.58/lib.mongooseim.logs/run.2020-04-10_02.19.59/suite.log.html)

**Pull request merged**